driver/src/main/scala/api/MongoConnection.scala

```diff
@@ -412,27 +412,48 @@ object MongoConnection {
 
       // ---
 
+      val (unsupportedKeys, options) = opts
       if (useful.indexOf("@") == -1) {
         val (db, hosts) = parseHostsAndDbName(useful)
-        val (unsupportedKeys, options) = opts
 
-        ParsedURI(hosts, options, unsupportedKeys, db, None)
+        options.authMode match {
+          case X509Authentication => ParsedURI(hosts, options, unsupportedKeys, db,
+            Some(Authenticate(db.getOrElse(throw new URIParsingException(s"Could not parse URI '$uri': authentication information found but no database name in URI")), "", "")))
+          case _ => ParsedURI(hosts, options, unsupportedKeys, db, None)
+        }
       } else {
         val WithAuth = """([^:]+):([^@]*)@(.+)""".r
+        val WithAuthNoPassword = """([^:]+)@(.+)""".r
```

Comment:
For me doesn't match the URL scheme with `mongodb://`

Reply:
@cchantep do you mean that password should be supplied even if not required? For x509 user is needed but private key is used instead of a password effectively

Reply:
```scala
val WithAuth = """([^:]+)(|:[^@]*)@(.+)""".r

"user@foo/db" match { case WithAuth(a, b, c) => println(s"OK: '$a' '${b drop 1}' '$c'"); case _ => println("KO") }
// --> OK: 'user' '' 'foo/db'

"user:pwd@foo/db" match { case WithAuth(a, b, c) => println(s"OK: '$a' '${b drop 1}' '$c'"); case _ => println("KO") }
// --> OK: 'user' 'pwd' 'foo/db'
```

Reply:
I'm not sure your `password.drop(1)` is immediately understandable but I've made a similar change that uses your regex